### PR TITLE
Sliding Window Maximum in JavaScript

### PR DESCRIPTION
**Pull request by Coder-Srinivas**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Sliding Window Maximum is a famous problem where we are given a list of integers
 nums and an integer k, return the maximum values of each sublist of length k.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue : #2304

### CODE REVIEW

**Comment by Coder-Srinivas** [comment]
@siddharth25pandey @HarshCasper I have added the code for sliding window maximum in JavaScript. You can test it [here](https://binarysearch.com/problems/Sliding-Window-Max). Please let me know if any changes are required. 😊

File: JavaScript/cp/SlidingWindowMaximum.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        /* Pushing the new maximum to the array */
+        array[i-k+1] = nums[window[0]];
+    }
+
+    /* Returning the answer */
+    return array;
+}
+
+
+console.log(slidingWindowMaximum([3, 2, 1, 2, 3, 7, 1, 3, 1, 5, 3], 4))
```
Comment (HarshCasper) [inline JavaScript/cp/SlidingWindowMaximum.js]: Add dynamic input

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
Closing this PR due to inactivity.

**Comment by Coder-Srinivas** [comment]
@HarshCasper I have added dynamic input using the readline() module. Please let me know if any changes are required.

File: JavaScript/cp/SlidingWindowMaximum.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+Example 2:
+
+Input:
+
+nums = [3, 2, 1, 2, 3]
+k = 5
+
+Output:
+[3]
```
Comment (ankitaggarwal23) [inline JavaScript/cp/SlidingWindowMaximum.js]: Write a proper display message how you are reading nums , k and printing output. 
Remove the extra testcase one is enough.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Coder-Srinivas** [comment]
@ankitaggarwal23 Made the changed requested.

File: JavaScript/cp/SlidingWindowMaximum.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+Example 1:
+
+Input:
+
+nums = [3, 2, 1, 2, 3]
+k = 2
+
+Output:
+
+[3, 2, 2, 3]
```
Comment (ankitaggarwal23) [inline JavaScript/cp/SlidingWindowMaximum.js]: Write the I/p and O/p the way your code is producing

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: JavaScript/cp/SlidingWindowMaximum.js (hunk 3 above)
Comment (Coder-Srinivas) [reply JavaScript/cp/SlidingWindowMaximum.js]: @ankitaggarwal23 Can you please explain the above comment? Should I write an explanation for the given input? Like how the code runs for the input?

Comment (ankitaggarwal23) [reply JavaScript/cp/SlidingWindowMaximum.js]: No It's not like that I'm saying the way your are getting the output after you compile and run that same way you have to write.

**Comment by Coder-Srinivas** [comment]
@ankitaggarwal23 Made the changes requested.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**